src/backend/identities.rs

```diff
@@ -555,6 +580,120 @@ impl AppState {
                     app_state_update,
                 }
             }
+            IdentityTask::AddPrivateKeys(ref private_key_strings) => {
+                let loaded_identity_lock = self.loaded_identity.lock().await;
+                let identity_id = &loaded_identity_lock
+                    .clone()
+                    .expect("An identity should be loaded")
+                    .id();
+                drop(loaded_identity_lock);
+
+                let result = Self::add_identity_with_private_keys_as_strings(
+                    &self,
+                    identity_id,
+                    &private_key_strings,
+                    sdk,
+                )
+                .await;
+
+                if result.is_err() {
+                    return BackendEvent::TaskCompleted {
+                        task: Task::Identity(task),
+                        execution_result: Err(format!(
+                            "Failed to add identity with private keys: {}",
+                            result.unwrap_err()
+                        )),
+                    };
+                }
```

Comment:
`if let Err(e) = .. { return *use e* }` seems to be a better fit instead of `unwrap_err`